contracts/ligo/src/english_auction/english_auction_tez_permit.mligo

```diff
@@ -0,0 +1,100 @@
+#include "./english_auction_tez.mligo"
+
+type permit = 
+  [@layout:comb]
+  {
+    signerKey: key;
+    signature: signature;
+    paramHash: bytes;
+  }
+
+type permit_storage = 
+    [@layout:comb]
+  {
+    auction_storage : storage;
+    counter : nat;
+  }
+
+type permit_config_param =
+  [@layout:comb]
+  {
+    config : configure_param;
+    optional_permit : permit option;
+  } 
+
+type permit_auction_entrypoints = 
+  | Auction of auction_without_configure_entrypoints
+  | Permit_config of permit_config_param list
+
+type permit_return = operation list * permit_storage
+
+let address_from_key (key : key) : address =
+  let a = Tezos.address (Tezos.implicit_account (Crypto.hash_key key)) in
+  a
+
+let config_storage_with_permit (p, permit_storage : permit_config_param * permit_storage)  : permit_storage = begin
+  match p.optional_permit with 
+    | None -> 
+      let auction_storage = configure_auction_storage(p.config, permit_storage.auction_storage) in
+      {permit_storage with auction_storage = auction_storage; counter = permit_storage.counter + 1n}
+    | Some permit -> 
+        let unsigned : bytes = ([%Michelson ({| { SELF; ADDRESS; CHAIN_ID; PAIR; PAIR; PACK } |} : nat * bytes -> bytes)] (permit_storage.counter, permit.paramHash) : bytes) in
```

Comment:
Consider breaking it into two functions for readability: `check_permission` and ` configure_auction`. Maybe you can reuse `configure_auction_storage` function from `english_auction_tez.mligo` if you make `seller` an explicit parameter